frontend/lib/src/components/widgets/DataFrame/columns/cells/LinkCell.tsx

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as React from "react"
+
+import {
+  CustomCell,
+  CustomRenderer,
+  getMiddleCenterBias,
+  measureTextCached,
+  blend,
+  GridCellKind,
+} from "@glideapps/glide-data-grid"
+import UriOverlayEditor from "./UriOverlayEditor"
+
+const UNDERLINE_OFFSET = 5
+
+export interface LinkCellProps {
+  readonly kind: "link-cell"
+  readonly href: string
+  readonly displayText?: string
+}
+
+export type LinkCell = CustomCell<LinkCellProps>
+
+function onClickSelect(
+  e: Parameters<NonNullable<CustomRenderer<LinkCell>["onSelect"]>>[0]
+): string | undefined {
+  const canvas = document.createElement("canvas")
+  const ctx = canvas.getContext("2d", { alpha: false })
+  if (ctx === null) return
+
+  const { posX: hoverX, bounds: rect, cell, theme } = e
+  const font = `${theme.baseFontStyle} ${theme.fontFamily}`
+  ctx.font = font
+
+  const { href, displayText } = cell.data
+
+  const rectHoverX = rect.x + hoverX
+
+  const textWidth =
+    ctx.measureText(displayText || href).width +
+    theme.cellHorizontalPadding * 2
```

Comment:
I think we can remove the horizontal padding here, or? Adding the padding here for the click action makes it more complicated to open the details/editor overlay.

```suggestion
    ctx.measureText(displayText || href).width
```

Reply:
It would be even better if we could also restrict the click to the height of the text.. but I think this is a lot more complicated, so might not be worth it.

Reply:
Yeah, I should have thought critically about that before just copy/pasting it from the determined library. Clicking on the text itself and not the right side padding should invoke the action.

Removing the padding from the `measure` method does reduce the width of my example table, but if that's fine with us then so be it.

before:
![image](https://github.com/streamlit/streamlit/assets/123410431/ea646136-1a6d-4fae-b933-259dc9eb4fae)
after:
![image](https://github.com/streamlit/streamlit/assets/123410431/ef8aae62-8463-4d9e-b6ea-b83d26c8f512)

Reply:
I also noticed a slight bug where the "bounding box" doesn't match up exactly with the text even after I removed the horizontal padding. `rect.x` is actually 0, but it should account for the cell padding. I think the correct logic for the hover is this:
```
const textStartX = rect.x + theme.cellHorizontalPadding
    const isHovered =
      rectHoverX > textStartX && rectHoverX < textStartX + rect.width
```
I tested this out and the click behavior now aligns well with my mouse position

https://github.com/streamlit/streamlit/assets/123410431/802f451c-d6a3-463d-a53c-784a57939113

Reply:
> Removing the padding from the measure method does reduce the width of my example table, but if that's fine with us then so be it.

There are two uses of `measureText` and `theme.cellHorizontalPadding * 2`. The one in `onClickSelect` should impact the column width since it is determined by the implementation in `measure` below, or?

Reply:
Not sure I understand your comment, but I would think that the `measure` method should keep the horizontal padding * 2, to account for both left and right padding, and the hover on the text should not account for that padding (other than to know when the text actually starts). Does that sound right to you? So measure would be:
```
measure: (ctx, cell, theme) => {
    const { href, displayText } = cell.data
    if (isNullOrUndefined(href)) return 0

    return (
      ctx.measureText(displayText || href).width +
      theme.cellHorizontalPadding * 2
    )
  },
  ```

and within the onClick the text width would be calculated without any extra padding:
`const textWidth = ctx.measureText((displayText || href) ?? "").width`

Reply:
yep, that sounds right for me 👍